### PR TITLE
[Cloud Run] Markdown sample

### PR DESCRIPTION
**Pull request by averikitsch**
Sample for tutorial:
https://cloud.google.com/run/docs/tutorials/secure-services

- [x] Tests pass (`mvn -P lint clean verify`)?
- [x] `pom.xml` parent set to latest `shared-configuration`
- [x] Appropriate changes to README are included in PR
- [x] API's need to be enabled to test (tell us)
- [x] Environment Variables need to be set (ask us to set them)

### CODE REVIEW

File: run/markdown-preview/editor/pom.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright 2019 Google LLC
```
Comment (lesv) [inline run/markdown-preview/editor/pom.xml:3]: If it's new, it probably should be 2020, also all the white space is missing.

File: run/markdown-preview/editor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.example.cloudrun</groupId>
+  <artifactId>editor</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <!--  The parent pom defines common style checks and testing strategies for our samples.
+	Removing or replacing it should not affect the execution of the samples in anyway.  -->
+  <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.0.15</version>
```
Comment (lesv) [inline run/markdown-preview/editor/pom.xml:24]: 1.0.17

File: run/markdown-preview/editor/pom.xml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>jib-maven-plugin</artifactId>
+        <version>2.1.0</version>
+        <configuration>
+          <to>
+            <image>gcr.io/PROJECT_ID/editor</image>
```
Comment (lesv) [inline run/markdown-preview/editor/pom.xml:86]: I hope you mention this in a README

File: run/markdown-preview/editor/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2019 Google LLC
```
Comment (lesv) [inline run/markdown-preview/editor/src/main/resources/application.properties:1]: 2020; whitespace ?

File: run/markdown-preview/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Cloud Run Markdown Preview Sample
```
Comment (lesv) [inline run/markdown-preview/README.md:1]: You might wish to mention any files they must modify to get this to work.  Such as pom.xml

File: run/markdown-preview/editor/src/main/resources/templates/index.html [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+      .catch((err) => {
+        console.log('Render Text: ' + err.message);
+        preview.innerHTML = '<h3><i aria-hidden="true" class="material-icons">error</i>Render Error</h3>\n<p>' + err.message + '</p>';
+      })
+      .finally(() => lp.close())
+    }
+
+    document.querySelector('.editor-button').addEventListener('click', listener);
+    window.addEventListener('load', listener);
+  </script>
```
Comment (lesv) [inline run/markdown-preview/editor/src/main/resources/templates/index.html:109]: Do we have any CSRF or XSRF protection on this?

File: run/markdown-preview/renderer/pom.xml [hunk 7]
```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright 2019 Google LLC
```
Comment (lesv) [inline run/markdown-preview/renderer/pom.xml:3]: 2020; whitespace

File: run/markdown-preview/renderer/pom.xml [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>jib-maven-plugin</artifactId>
+        <version>2.1.0</version>
+        <configuration>
+          <to>
+            <image>gcr.io/PROJECT_ID/renderer</image>
```
Comment (lesv) [inline run/markdown-preview/renderer/pom.xml:97]: Mention in README

File: run/markdown-preview/renderer/src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2019 Google LLC
```
Comment (lesv) [inline run/markdown-preview/renderer/src/main/resources/application.properties:1]: 2020; whitespace

File: run/markdown-preview/editor/src/main/resources/templates/index.html (hunk 6 above)
Comment (averikitsch) [reply run/markdown-preview/editor/src/main/resources/templates/index.html:109]: The service uses the metadata server to add a token to make an authenticated request to the renderer service. I also sanitize the rendered HTML.

File: run/markdown-preview/editor/pom.xml (hunk 1 above)
Comment (averikitsch) [reply run/markdown-preview/editor/pom.xml:3]: done.

File: run/markdown-preview/editor/pom.xml (hunk 2 above)
Comment (averikitsch) [reply run/markdown-preview/editor/pom.xml:24]: done.

File: run/markdown-preview/editor/src/main/resources/application.properties (hunk 4 above)
Comment (averikitsch) [reply run/markdown-preview/editor/src/main/resources/application.properties:1]: done.

File: run/markdown-preview/renderer/pom.xml (hunk 7 above)
Comment (averikitsch) [reply run/markdown-preview/renderer/pom.xml:3]: done.

File: run/markdown-preview/renderer/src/main/resources/application.properties (hunk 9 above)
Comment (averikitsch) [reply run/markdown-preview/renderer/src/main/resources/application.properties:1]: done.

File: run/markdown-preview/renderer/pom.xml (hunk 8 above)
Comment (averikitsch) [reply run/markdown-preview/renderer/pom.xml:97]: done.

File: run/markdown-preview/README.md (hunk 5 above)
Comment (averikitsch) [reply run/markdown-preview/README.md:1]: done.

File: run/markdown-preview/editor/pom.xml (hunk 3 above)
Comment (averikitsch) [reply run/markdown-preview/editor/pom.xml:86]: done

**Pull request merged**